recipe-server/client/control/state/approvalRequests/actions.js

```diff
@@ -0,0 +1,97 @@
+import { getRequest } from './selectors';
+
+import {
+  APPROVAL_REQUEST_FETCH,
+  APPROVAL_REQUEST_FETCH_FAILURE,
+  APPROVAL_REQUEST_FETCH_SUCCESS,
+  APPROVAL_REQUEST_RECEIVE,
+  APPROVAL_REQUESTS_FETCH,
+  APPROVAL_REQUESTS_FETCH_FAILURE,
+  APPROVAL_REQUESTS_FETCH_SUCCESS,
+} from '../action-types';
+
+import apiFetch from '../../utils/apiFetch';
+
+
+function fetchApprovalRequestSuccess(dispatch, requestId, approvalRequest) {
+  dispatch({
+    type: APPROVAL_REQUEST_FETCH_SUCCESS,
+    requestId,
+  });
+
+  dispatch({
+    type: APPROVAL_REQUEST_RECEIVE,
+    approvalRequest,
+  });
+}
+
+
+function fetchApprovalRequestFailure(dispatch, requestId, error) {
+  dispatch({
+    type: APPROVAL_REQUEST_FETCH_FAILURE,
+    error,
+    requestId,
+  });
+}
+
+
+export function fetchApprovalRequest(pk) {
+  return (dispatch, getState) => {
+    const requestId = `fetch-${pk}`;
+    const request = getRequest(getState(), requestId);
+
+    if (request.loading) { return true; }
```

Comment:
I'm not sure how `Immutable#getIn` works - will it always return an object? If not, we will need to safeguard this conditional (`if (request && request.loading){ ...`)